File: Binary Number with Alternating Bits (693)/693.py
# Runtime: 20 ms
# Beats 100% of Python submissions

class Solution(object):
    def hasAlternatingBits(self, n):
        """
        :type n: int
        :rtype: bool
        """
        while n != 0:
            prev_bit = n & 1
            n = n >> 1
            curr_bit = n & 1
            if prev_bit == curr_bit:
                return False
            
        return True

File: Binary Number with Alternating Bits (693)/693.java
// Runtime: 7 ms
// Beats 100% of Java submissions

class Solution {
    public boolean hasAlternatingBits(int n) {
        while (n != 0) {
            int prevBit = n & 1;
            n = n >> 1;
            int currBit = n & 1;
            if (prevBit == currBit) {
                return false;
            }
        }
        
        return true;
    }
}